src/snekmate/auth/mocks/Ownable2StepMock.vy

```diff
@@ -0,0 +1,50 @@
+# pragma version ~=0.4.0b5
+"""
+@title Ownable2Step Module Reference Implementation
+@custom:contract-name Ownable2StepMock
+@license GNU Affero General Public License v3.0 only
+@author pcaversaccio
+"""
+
+
+# @dev We import and initialise the `Ownable` module.
+from .. import Ownable as ow
+initializes: ow
+
+
+# @dev We import and initialise the `Ownable2Step` module.
+from .. import Ownable2Step as o2
+initializes: o2[ownable := ow]
+
+
+# @dev We export (i.e. the runtime bytecode exposes these
+# functions externally, allowing them to be called using
+# the ABI encoding specification) all `external` functions
+# from the `Ownable2Step` module.
+# @notice Please note that you must always also export (if
+# required by the contract logic) `public` declared `constant`,
+# `immutable`, and state variables, for which Vyper automatically
+# generates an `external` getter function for the variable.
+exports: (
+    ow.owner,
+    o2.pending_owner,
+    o2.transfer_ownership,
+    o2.accept_ownership,
+    o2.renounce_ownership,
+)
+
+
+@deploy
+@payable
+def __init__():
+    """
+    @dev To omit the opcodes for checking the `msg.value`
+         in the creation-time EVM bytecode, the constructor
+         is declared as `payable`.
+    @notice The `owner` role will be assigned to
+            the `msg.sender`.
+    """
+    # The following line assigns the `owner`
+    # to the `msg.sender`.
+    ow.__init__()
+    o2.__init__()
```

Comment:
is it needed to call the `o2` init if it just passes?

Reply:
Yes it's needed otherwise it would throw with:

```vy
vyper.exceptions.InitializerException: not initialized!

  (hint: add `o2.__init__()` to your `__init__()` function)

  contract "src\snekmate\auth\mocks\Ownable2StepMock.vy:17", line 17:0 
       16 from .. import Ownable2Step as o2
  ---> 17 initializes: o2[ownable := ow]
  --------^
       18
```

Reply:
It's my opinionated design to make constructor `payable` to save the deployment costs, and thus I add them to the snekmate contracts. It wouldn't throw tho if I didn't add an `__init__` function in `Ownable2Step`.

Reply:
gotcha, wrt to the `__init__` function in `Ownable2Step`: https://github.com/pcaversaccio/snekmate/pull/219#discussion_r1539377119